File: tests/fixtures/python/even.py
"""Case d (D4): cross-file call cycle (mutual recursion) with odd.py. The
`from ... import` form makes the call site a bare identifier, so this
resolves via R5 (project-unique bare name) rather than R1/R2 — see
../README.md "Contract interpretations" (same reasoning as the TypeScript
fixture's even.ts/odd.ts)."""
from odd import is_odd


def is_even(n):
    if n == 0:
        return True
    return is_odd(n - 1)

File: tests/fixtures/typescript/src/even.ts
/**
 * Case d (D4): cross-file call cycle (mutual recursion) with odd.ts. The
 * named import (rather than a namespace import) makes the call site a bare
 * identifier, so this resolves via R5 (project-unique bare name), not R1 —
 * see ../README.md "Contract interpretations" (R4 only consults import
 * facts for Rust in v1, so a bare name reached through any other language's
 * import falls through to R5 whenever it happens to be project-unique).
 */
import { isOdd } from './odd';

export function isEven(n: number): boolean {
  return n === 0 ? true : isOdd(n - 1);
}
